libsel4vm/src/guest_memory.c

```diff
@@ -394,13 +395,13 @@ vm_memory_reservation_t *vm_reserve_anon_memory(vm_t *vm, size_t size,
         return NULL;
     }
 
-    allocable_region = find_allocable_anon_region(vm, ROUND_UP(size, BIT(seL4_PageBits)));
+    allocable_region = find_allocable_anon_region(vm, ROUND_UP(size, BIT(seL4_PageBits)), align);
     if (!allocable_region) {
         ZF_LOGE("Failed to reserve anon memory: No anonymous memory available to cater reservation size");
         return NULL;
     }
 
-    reservation_addr = allocable_region->alloc_addr;
+    reservation_addr = ROUND_UP(allocable_region->alloc_addr, (uintptr_t) align);
```

Comment:
I think you will also will need to update line 426 from
`allocable_region->alloc_addr += ROUND_UP(size, BIT(seL4_PageBits))` 
to
`allocable_region->alloc_addr = reservation_addr + ROUND_UP(size, BIT(seL4_PageBits))`
To make sure the next sub-allocation in this anon region won't overlap with your current aligned allocation 
_(Sorry Github doesn't seem to allow me to comment on the actual line)_

Reply:
Oh wow, absolutely, I totally overlooked this important detail. Thanks for that, I added your suggestion verbatim and force-pushed the branch.